src/rpc/core_rpc_server.cpp

```diff
@@ -3300,6 +3300,7 @@ namespace cryptonote { namespace rpc {
   //------------------------------------------------------------------------------------------------------------------------------
   STORAGE_SERVER_PING::response core_rpc_server::invoke(STORAGE_SERVER_PING::request&& req, rpc_context context)
   {
+    m_core.ss_version = {static_cast<uint16_t>(req.version_major), static_cast<uint16_t>(req.version_minor), static_cast<uint16_t>(req.version_patch)};
```

Comment:
Any reason not to just make the `.version_major` etc. fields just uint16_t's in the first place?